### DISCUSSION TITLE
2nd USB device with CH32V30x

### DISCUSSION THREAD

**Discussion by lyusupov** [adafruit/Adafruit_TinyUSB_Arduino#470] [General]
Current implementation of [Adafruit_TinyUSB_Arduino](https://github.com/adafruit/Adafruit_TinyUSB_Arduino) gives one USB port available for CH32 VXXX MCUs.
However, some of the devices have 2 USB ports exposed, see an illustration below.
Could you, please, provide an example on how to use the 2nd USB port as well ?

![](https://raw.githubusercontent.com/yym36100/RISC-V_CH32V307VCT6/refs/heads/main/Capture.JPG)

**Comment by hathach** [comment]
Running 2 devices is not supported at the moment

**Reply by lyusupov** [reply]
Thank you for explaining the implementation limitation!
